pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -30,30 +30,33 @@ import (
 
 const primaryPodInterfaceName = "eth0"
 
-type VMNetworkConfigurator struct {
+type NetworkingConfigurator interface {
+	Setup() error
+}
+
+type InfraNetworkingConfigurator struct {
 	vmi          *v1.VirtualMachineInstance
 	handler      netdriver.NetworkHandler
 	cacheFactory cache.InterfaceCacheFactory
+	launcherPID  int
+	nicMaker     NicGenerator
 }
 
-func newVMNetworkConfiguratorWithHandlerAndCache(vmi *v1.VirtualMachineInstance, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return &VMNetworkConfigurator{
-		vmi:          vmi,
-		handler:      handler,
-		cacheFactory: cacheFactory,
-	}
-}
-
-func NewVMNetworkConfigurator(vmi *v1.VirtualMachineInstance, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return newVMNetworkConfiguratorWithHandlerAndCache(vmi, &netdriver.NetworkUtilsHandler{}, cacheFactory)
+type NetworkingSpecGenerator struct {
+	vmi          *v1.VirtualMachineInstance
+	handler      netdriver.NetworkHandler
+	cacheFactory cache.InterfaceCacheFactory
+	domXML       api.Domain
+	nicMaker     NicGenerator
 }
 
-func (v VMNetworkConfigurator) getNICs() ([]podNIC, error) {
-	return v.getNICsWithLauncherPID(nil)
-
+type NicGenerator struct {
```

Comment:
Let's rename this `NicFactory`

Reply:
What is the advantage ? E.g. what does a generator do vs what does a factory do ?

Reply:
The Generator word is used for things like code generators, while Factory word is more often used for kind of constructors, for example we have the pattern of "factory methods" and this is what we have before moving the getNIC code to a struct.

Reply:
I wonder what kind of code generation would anyone expect out of this.

I'll rename it, I can live with either name.